### DISCUSSION TITLE
Common Core Ontology Software Libraries

### DISCUSSION THREAD

**Discussion by daniel-raimi-zlatic** [CommonCoreOntology/CommonCoreOntologies#322] [Help]
I have a need for software libraries using common core's representation of information entities. Later on in this project, I foresee wanting software libraries for more common core ontologies. Are there any open source, or closed source, libraries that I could use, and if not, are any planned? Python or Java would be preferable.

**Comment by mark-jensen** [comment]
Could you describe in more detail or provide an example of the kinds of software libraries you're referring to?

**Comment by daniel-raimi-zlatic** [comment]
Sure. I'm looking for software libraries containing tools that the developer can utilize to more easily generate graphs with CCO recommended modeling patterns. I'd also like the software library to do useful, common things with graphs that follow CCO modeling patterns.

For example, here are the CCO use-cases that I've currently written in python:

I made classes that are representations of "information". These objects have properties defined using the rdflib library that represent resources relating to information - its basic properties are basically a subject entity, an information content entity, an information bearing entity, and a value Literal. These classes also define methods that for instance: 1. write triples for this information with the modeling pattern given in "Modeling Information with the Common Core Ontologies 1.0" to an rdflib Graph, 2. gets the appropriate CCO data property to use for Literal based on the Literal data type, 3. gets the appropriate "subject_of" property for the information content entity, 4. searches an rdflib Graph for ties between all subjects and their tokenizations given some information content entity type.
